File: src/main/java/net/sf/rails/game/state/BooleanChange.java
package net.sf.rails.game.state;

/**
 * Change associated with BooleanState
 */

final class BooleanChange extends Change {

    private final BooleanState state;
    private final boolean newValue;
    private final boolean oldValue;
    
    BooleanChange(BooleanState state,  boolean newValue) {
        this.state = state;
        this.newValue = newValue;
        this.oldValue = state.value();
        super.init(state);
    }

    @Override void execute() {
        state.change(newValue);
    }

    @Override void undo() {
        state.change(oldValue);
    }

    @Override BooleanState getState() {
        return state;
    }

    @Override
    public String toString() {
        return "Change for " + state + ": From " + oldValue + " to " + newValue; 
    }
        
}

File: dotnet/RailsLib.Net/Net/Game/State/BooleanChange.cs
using System;

/**
 * Change associated with BooleanState
 */
namespace GameLib.Net.Game.State
{
    public sealed class BooleanChange : Change
    {
        private readonly BooleanState state;
        private readonly bool newValue;
        private readonly bool oldValue;

        public BooleanChange(BooleanState state, bool newValue)
        {
            this.state = state;
            this.newValue = newValue;
            this.oldValue = state.Value;
            base.Init(state);
        }

        public override void Execute()
        {
            state.Change(newValue);
        }

        public override void Undo()
        {
            state.Change(oldValue);
        }

        override public GameState GameState
        {
            get
            {
                return state;
            }
        }

        override public String ToString()
        {
            return "Change for " + state + ": From " + oldValue + " to " + newValue;
        }
    }
}
